core/templates/dev/head/domain/statistics/ImprovementActionButtonObjectFactory.ts

```diff
@@ -32,18 +32,22 @@ export class ImprovementActionButton {
   /**
    * @constructor
    * @param {string} text - The text displayed on the button.
-   * @param {callback} actionFunc - Function to run when the button is clicked.
    * @param {string} [cssClass=btn-default] - The CSS class to render the button
    *    with.
+   * @param {callback} actionFunc - Function to run when the button is clicked.
+   * @param {callback} enabledFunc - Function which returns whether this button
+   *    should be enabled and clickable.
    */
   // TODO(#7165): Replace 'any' with the exact type. This has been kept as
   // 'any' because '_actionFunc' is a function with varying return types
   // depending upon the arguments paased to the constructor of
   // 'ImprovementActionButton'.
-  constructor(text: string, cssClass: string, actionFunc: any) {
+  constructor(
+      text: string, cssClass: string, actionFunc: any, enabledFunc: any) {
```

Comment:
Will it be possible to add the type of `enabledFunc`?

Reply:
Done.